lib/plugin/opsgenie/plugindata.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2020-2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package opsgenie
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+// PluginData is a data associated with access request that we store in Teleport using UpdatePluginData API.
+type PluginData struct {
+	RequestData
+	OpsgenieData
+}
+
+// Resolution stores the resolution (approved, denied or expired) and its reason.
+type Resolution struct {
+	Tag    ResolutionTag
+	Reason string
+}
+type ResolutionTag string
+
+const Unresolved = ResolutionTag("")
+const ResolvedApproved = ResolutionTag("approved")
+const ResolvedDenied = ResolutionTag("denied")
+const ResolvedExpired = ResolutionTag("expired")
```

Comment:
I guess these types and const should be declared at a higher level because they are /will be re-used across plugins